### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by joesus**
## Checklist

- [x] I've read the [Contributing Guidelines](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I've completed the [Contributor License Agreement](https://developers.facebook.com/opensource/cla)
- [x] I've ensured that all existing tests pass and added tests (when/where necessary)
- [x] I've ensured that my code lints properly: (`swiftlint && swiftlint autocorrect --format`)
- [x] I've updated the documentation (when/where necessary) and [Changelog](CHANGELOG.md) (when/where necessary)
- [x] I've added the proper label to this pull request (e.g. `bug` for bug fixes)

## Pull Request Details

Enables conversion of string backed enums to their localized variants.
Sets up pattern for testing future localized string enums

## Test Plan
All tests should pass.

### CODE REVIEW

File: Sources/Core/ErrorRecoveryStrings.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+enum ErrorRecoveryStrings: String, CaseIterable, Localizable {
+  case ok = "ErrorRecovery.OK"
+  case cancel = "ErrorRecovery.Cancel"
+  case transientSuggestion = "ErrorRecovery.Transient.Suggestion"
```
Comment (codytwinton) [inline Sources/Core/ErrorRecoveryStrings.swift]: Why the extra namespace?

```suggestion
  case transientSuggestion = "ErrorRecovery.TransientSuggestion"
```

File: Sources/Core/ErrorRecoveryStrings.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  case loginRecoverySuggestion = "ErrorRecovery.Login.Suggestion"
```
Comment (codytwinton) [inline Sources/Core/ErrorRecoveryStrings.swift]:
```suggestion
  case loginRecoverySuggestion = "ErrorRecovery.LoginSuggestion"
```

**Comment by joesus** [comment]
Discussed with @codytwinton about wanting to do some more digging into the reasons behind the existing localized string naming conventions before changing them. Merging for now since will be easy to revisit.

File: Sources/FacebookCoreTests/Helpers/LocalizedStringValidation.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+@testable import FacebookCore
+import XCTest
+
+// Uses an arbitrary class that is located in the SDK's bundle
+private let bundle = Bundle(for: AccessTokenWallet.self)
+
+func validateLocalizedStrings<T: CaseIterable & RawRepresentable>(stringsEnum: T.Type, _ file: StaticString = #file, _ line: UInt = #line) where T.RawValue == String {
+  // make sure there is entry for each case in a localizable type
```
Comment (robtimp) [inline Sources/FacebookCoreTests/Helpers/LocalizedStringValidation.swift]: Oh good! I see things like this in production apps far too often (this is Netflix):

Screen Shot 2019-03-14 at 10 28 48 AM

**Pull request merged**